.evergreen/resync-specs.sh

```diff
@@ -56,10 +56,14 @@ cpjson () {
     cd "$SPECS"/source/$1
     find . -name '*.json' | grep -Ev "${BLOCKLIST}" | cpio -pdm \
     $PYMONGO/test/$2
-    printf "\nIgnored files for ${PWD}"
-    printf "\n%s\n" "$(diff <(find . -name '*.json' | sort) \
+    printf "\nIgnored files for ${PWD}\n"
+    IGNORED_FILES="$(printf "\n%s\n" "$(diff <(find . -name '*.json' | sort) \
     <(find . -name '*.json' | grep -Ev "${BLOCKLIST}" | sort))" | \
-    sed -e '/^[0-9]/d' | sed -e 's|< ./||g'
+    sed -e '/^[0-9]/d' | sed -e 's|< ./||g' )"
+    printf "%s\n" $IGNORED_FILES
+    cd "$PYMONGO"/test/$2
+    printf "%s\n" $IGNORED_FILES | xargs git checkout master
```

Comment:
This is mainly tweaks to the functionality of `resync_specs.sh` that I discovered while implementing this ticket. The main issue is just that when it "ignores" a file, that means it does not copy it over. However, because we delete all files in the directory prior, that means that now the file has been deleted. The solution is to checkout those ignored files after copying to ensure they will not show in the PR rather than showing up as a deletion.